File: admin/src/main/java/com/project/admin/controller/dto/response/release/GitHubReleaseResponse.java
package com.project.admin.controller.dto.response.release;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.project.admin.service.dto.githubrelease.GitHubReleaseDetailResult;
import java.time.LocalDateTime;

public record GitHubReleaseResponse(
        String id,
        String techStack,
        String tagName,
        String name,
        String category,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
        LocalDateTime publishedAt,
        Boolean prerelease,
        String publisher,
        String body) {


    public static GitHubReleaseResponse of(GitHubReleaseDetailResult result) {

        return new GitHubReleaseResponse(
                result.id(),
                result.techStack(),
                result.tagName(),
                result.name(),
                result.category(),
                result.publishedAt(),
                result.prerelease(),
                result.publisher(),
                result.body()

        );
    }
}

File: batch/src/main/kotlin/com/project/batch/remote/dto/GithubReleaseResponse.kt
package com.project.batch.remote.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class GithubReleaseResponse(
    val id: Long,
    @JsonProperty("tag_name") val tagName: String,
    val name: String?,
    val body: String?,
    @JsonProperty("published_at") val publishedAt: String,
    val prerelease: Boolean,
    val draft: Boolean,
)